Networking/Networking/Remote/SiteVisitStatsRemote.swift

```diff
@@ -48,3 +53,37 @@ private extension SiteVisitStatsRemote {
         static let statFields: String = "stat_fields"
     }
 }
+
+/// An error that occurs at site visit stats networking layer.
+/// API documentation of possible errors:
+/// https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/
+///
+public enum SiteVisitStatsRemoteError: Error {
+    case statsModuleDisabled
+    case unknown
+
+    private enum ErrorIdentifiers {
+        static let invalidBlog: String = "invalid_blog"
+    }
+
+    private enum ErrorMessages {
+        static let statsModuleDisabled: String = "This blog does not have the Stats module enabled"
+    }
+
+    init(error: Error) {
```

Comment:
My first inclination would be making this method static, as in a [static factory method](https://stackify.com/static-factory-methods/). But, to be completely honest, I am not sure that would actually bring too many benefits.

Reply:
I just read the example in the [static factory method](https://stackify.com/static-factory-methods/) post, and it seems like the benefits of the static factory method over initializer are:
- static factory method is a good way to provide different constructors for individual use cases, some with more arguments some with fewer --> in Swift, we have designated and convenience initializers
- it looks like the constructor in the example `TimeSpan` doesn't have a signature for each input argument, my guess is that the language doesn't support it. The static factory method provides a way to describe the argument(s) in the function signature --> in Swift, we can specify what each argument is in the signature with optional different external/internal naming

this is my interpretation of the post, I think Swift initializers address the issues in the example. Lemme know if I missed anything from the post / other benefits of static factory method!

Reply:
I think that's pretty much it. As you mentioned, in this case I am not sure it actually brings much benefit.